File: src/main/java/chat/teco/tecochat/member/presentation/controller/MemberController.java
package chat.teco.tecochat.member.presentation.controller;

import static org.springframework.http.HttpStatus.CREATED;

import chat.teco.tecochat.member.application.service.MemberService;
import chat.teco.tecochat.member.presentation.controller.request.SignUpRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
public class MemberController {

    private final MemberService memberService;

    @PostMapping("/members")
    ResponseEntity<Void> signUp(@RequestBody SignUpRequest signUpRequest) {
        memberService.signUp(signUpRequest.toServiceDto());
        return ResponseEntity.status(CREATED.value()).build();
    }
}

File: src/main/kotlin/chat/teco/tecochat/ui/MemberController.kt
package chat.teco.tecochat.ui

import chat.teco.tecochat.application.MemberData
import chat.teco.tecochat.application.MemberService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class MemberController(
    private val memberService: MemberService
) {

    @PostMapping("/members")
    fun signUp(@RequestBody memberData: MemberData): ResponseEntity<Void> {
        memberService.signUp(memberData)
        return ResponseEntity.status(HttpStatus.CREATED.value()).build()
    }
}
